v3_async_wip/tests/test_iothub_mqtt_client.py

```diff
@@ -731,8 +729,89 @@ def message(self):
         return models.Message("some payload")
 
     @pytest.mark.it(
-        "Awaits a publish to the telemetry topic using the MQTTClient, sending the given Message's JSON-converted payload"
+        "Awaits a publish to the telemetry topic using the MQTTClient, sending the given Message's payload converted to bytes"
+    )
+    @pytest.mark.parametrize(
+        "device_id, module_id",
+        [
+            pytest.param(FAKE_DEVICE_ID, None, id="Device Configuration"),
+            pytest.param(FAKE_DEVICE_ID, FAKE_MODULE_ID, id="Module Configuration"),
+        ],
+    )
+    async def test_mqtt_publish(self, mocker, client, device_id, module_id):
+        assert client._mqtt_client.publish.await_count == 0
+        client._device_id = device_id
+        client._module_id = module_id
+        message = models.Message(payload="some_payload")
+        base_topic = mqtt_topic.get_telemetry_topic_for_publish(device_id, module_id)
+        expected_topic = mqtt_topic.insert_message_properties_in_topic(
+            topic=base_topic,
+            system_properties=message.get_system_properties_dict(),
+            custom_properties=message.custom_properties,
+        )
+        expected_payload = message.payload.encode("utf-8")
+
+        assert client._mqtt_client.publish.await_count == 0
+        await client.send_message(message)
+
+        assert client._mqtt_client.publish.await_count == 1
+        assert client._mqtt_client.publish.await_args == mocker.call(
+            expected_topic, expected_payload
+        )
+        assert isinstance(expected_payload, bytes)
+
+    @pytest.mark.it(
+        "Derives the byte payload from the Message payload according to the Message's content encoding and content type properties"
+    )
+    @pytest.mark.parametrize(
+        "device_id, module_id",
+        [
+            pytest.param(FAKE_DEVICE_ID, None, id="Device Configuration"),
+            pytest.param(FAKE_DEVICE_ID, FAKE_MODULE_ID, id="Module Configuration"),
+        ],
     )
+    @pytest.mark.parametrize("content_encoding", ["utf-8", "utf-16", "utf-32"])
+    @pytest.mark.parametrize(
+        "content_type, payload, expected_str_payload",
+        [
+            pytest.param("text/plain", "some_text", "some_text", id="text/plain"),
+            pytest.param(
+                "application/json", {"some": "json"}, '{"some": "json"}', id="application/json"
```

Comment:
>"application/json", [](http://example.com/codeflow?start=16&length=19)

dumb corner case that is easy to screw up/break. You can json encode a string, and you get a string with that string in quotes.

Reply:
Not actually a corner case - that's desired behavior and is covered a little bit farther down in a separate test